magicctapipe/scripts/lst1_magic/semi_automatic_scripts/check_MAGIC_runs.py

```diff
@@ -0,0 +1,134 @@
+    result['MAGIC runs'] = result['MAGIC runs'].apply(fix_lists_and_convert)
+    return(result)
+
+def existing_files( tel_id, date, source, magic_runs ):
+
+    magic_runs = str(magic_runs)  
+    date_obj = datetime.strptime(date, '%Y%m%d')
+    date_obj += timedelta(days=1)
+    new_date = datetime.strftime(date_obj, '%Y%m%d')
+    YYYY = new_date[:4]
+    MM = new_date[4:6]
+    DD = new_date[6:8]
+    Y = f"_Y_"
+    
+    path = f"/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}" 
+    
+    if os.path.exists(path):
+        files = os.listdir(path)
+        count_with_source = 0  
+        count_with_run_id = 0
+            # Counter for files that include the source.  
+            # Counter for files that include the run_id.
+        for filename in files:
+            if date and source and Y in filename:
+                count_with_source += 1
+                if magic_runs in filename:
+                    count_with_run_id += 1
+        if count_with_source != 0 and count_with_run_id != 0:
+            print(f"{date}\t{source}\t{magic_runs}\t{count_with_run_id}")
+                    
+def missing_files( tel_id, date, source, magic_runs ):
+    
+    for runs in magic_runs:
+        run = str(runs)
+    
+    date_obj = datetime.strptime(date, '%Y%m%d')
+    date_obj += timedelta(days=1)
+    new_date = datetime.strftime(date_obj, '%Y%m%d')
+    YYYY = new_date[:4]
+    MM = new_date[4:6]
+    DD = new_date[6:8]
+    Y = f"_Y_"
+    
+    path = f"/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}" 
+    
+    if os.path.exists(path):
+        files = os.listdir(path)
+        count_with_source = 0  
+        count_with_run_id = 0
+        # Counter for files that include the source. We want to check if any file with the source was found. 
+        # Counter for files that include the run_id. We want to check if any file with the run_id was found. 
+        for filename in files:
+            if date and source and Y in filename:
+                count_with_source += 1
+                for runs in magic_runs:
+                   # run = str(runs)
+                    if run in filename:
+                        count_with_run_id += 1
+        if count_with_source == 0:  
+            if(tel_id == 1):
+                #Between 2022/09/04 - 2022/12/14 MAGIC 1 had a failure. Therefore we have to skip the range when we want to get information about missing files.
+                if(date<='20220904' or date>='20221214'):
+                    print(f"No files found containing the source '{source}' on {date}")
+                else:
+                    print(f"M1 failure. No files found containing the source '{source}' on {date}.")
+            if(tel_id == 2):
+                print(f"No files found containing the source '{source}' on {date}")
+        if count_with_source != 0 and count_with_run_id == 0:
+            if(date<'20220904' or date>'20221214'):
```

Comment:
this should be only for M1
in general I think it would be good to make a list of triples: telescope, date_min, date_max and loop over this list to see if the error is in one of the known bad regions